### PR TITLE
fix: use our node for eip-1559 gas estimation in all swappers

### LINKED ISSUE #5766: eip 1559 support in swapper

### PR DESCRIPTION
**Pull request by woodenfurniture**
## Description

Upgrades swappers to use our node for network fee estimates when building an unsigned tx. This ensures more accurate gas estimates as well as adding eip-1559 support across the board.

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #5766

Moderate risk of broken trades. Trade will fail if gas estimation is wrong.

> What protocols, transaction types or contract interactions might be affected by this PR?

Related to swappers (lifi, zrx, 1inch, thorchain (L1 and Longtail)

## Testing

- Ensure EIP-1559 fees are being used for supported wallets:
  * lifi
  * zrx
  * 1inch
  * thorchain (L1 and Longtail)

### Engineering

:point_up: 

### Operations

:point_up:

### CODE REVIEW

**Comment by woodenfurniture** [comment]
Current dependencies on/for this PR:
* `develop`
  * **PR #6153** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6153?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/6153?utm_source=stack-comment).

File: src/lib/swapper/swappers/ZrxSwapper/endpoints.ts [hunk 1]
```diff
@@ -49,16 +50,24 @@ export const zrxApi: SwapperApi = {
 
     if (zrxQuoteResponse.isErr()) throw zrxQuoteResponse.unwrapErr()
 
-    const { value, to, gasPrice, gas, data } = zrxQuoteResponse.unwrap()
+    const { value, to, data } = zrxQuoteResponse.unwrap()
+
+    const feeData = await getFees({
+      adapter: assertGetEvmChainAdapter(chainId),
+      data: data.toString(),
```
Comment (gomesalexandre) [inline src/lib/swapper/swappers/ZrxSwapper/endpoints.ts:57]: This should already be a string according to types
```suggestion
      data,
```

File: src/lib/swapper/swappers/ZrxSwapper/endpoints.ts [hunk 2]
```diff
@@ -49,16 +50,24 @@ export const zrxApi: SwapperApi = {
...
+      to,
+      value: value.toString(),
```
Comment (gomesalexandre) [inline src/lib/swapper/swappers/ZrxSwapper/endpoints.ts:59]: This should already be a string according to types

```suggestion
      value,
```

**Review by gomesalexandre — COMMENTED** [review]
Conceptually looks sane pending runtime testing

**Review by gomesalexandre — APPROVED** [review]
Tested locally with MM (which supports EIP1559 fees) and Ledger, which doesn't.

### MM

### 0x - EIP1559 is used ✅ 

image
image
image
image
image

### Li.Fi - EIP1559 is used ✅ 

image
image
image
image

### 1inch

Wasn't able to test this one because of current rps issues

### Ledger (which doesn't support EIP1559 fees)

#### 1inch

image
image
image

#### 0x

image
image
image

**Review by gomesalexandre — APPROVED** [review]
Retested locally with MM (which supports EIP1559 fees) and Ledger, which doesn't - 0x and Li.Fi this time around at 1inch has been proper disabled

### MM

### 0x - EIP1559 is used ✅ 

image
image
image


### Li.Fi - EIP1559 is used ✅ 

image
image
image



### Ledger (which doesn't support EIP1559 fees)

#### Li.Fi

image

image

image



#### 0x

image
image
image

**Pull request merged**